src/test/java/com/LuuQu/medicalclinic/service/AppointmentServiceTest.java

```diff
@@ -0,0 +1,229 @@
+        LocalDateTime startTime = LocalDateTime.now().plusDays(1).withMinute(0);
+        appointmentDto.setStartDate(startTime);
+        LocalDateTime endTime = LocalDateTime.now().plusDays(1).withMinute(0);
+        appointmentDto.setEndDate(endTime);
+
+        //when i then
+        assertThatThrownBy(() -> appointmentService.addAppointment(1L, appointmentDto))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Start date and end date shouldn't be the same");
+    }
+
+    @Test
+    void addAppointment_nonExistentDoctor_CorrectException() {
+        AppointmentDto appointmentDto = new AppointmentDto();
+        LocalDateTime startTime = LocalDateTime.now().plusDays(1).withMinute(0);
+        appointmentDto.setStartDate(startTime);
+        LocalDateTime endTime = LocalDateTime.now().plusDays(1).plusHours(1).withMinute(0);
+        appointmentDto.setEndDate(endTime);
+        when(doctorRepository.findById(1L)).thenReturn(Optional.empty());
+        //when i then
+        assertThatThrownBy(() -> appointmentService.addAppointment(1L, appointmentDto))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Doctor doesn't exist");
+    }
+
+    @Test
+    void addAppointment_doctorHaveAppointmentOnSetTime_CorrectException() {
+        AppointmentDto appointmentDto = new AppointmentDto();
+        LocalDateTime startTime = LocalDateTime.now().plusDays(1).withMinute(0);
+        appointmentDto.setStartDate(startTime);
+        LocalDateTime endTime = LocalDateTime.now().plusDays(1).plusHours(1).withMinute(0);
+        appointmentDto.setEndDate(endTime);
+        when(doctorRepository.findById(1L)).thenReturn(Optional.of(new Doctor()));
+        List<Appointment> list = new ArrayList<>();
+        list.add(new Appointment());
+        list.add(new Appointment());
+        when(appointmentRepository.getBetweenTime(1L, startTime, endTime)).thenReturn(list);
+        //when i then
+        assertThatThrownBy(() -> appointmentService.addAppointment(1L, appointmentDto))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Doctor already have appointment on set time");
+    }
+
+    @Test
+    void addAppointment_AppointmentAdded_ReturnAppointmentDto() {
+        AppointmentDto appointmentDto = new AppointmentDto();
+        LocalDateTime startTime = LocalDateTime.now().plusDays(1).withMinute(0);
+        appointmentDto.setStartDate(startTime);
+        LocalDateTime endTime = LocalDateTime.now().plusDays(1).plusHours(1).withMinute(0);
+        appointmentDto.setEndDate(endTime);
+        Doctor doctor = new Doctor();
+        doctor.setId(1L);
+        when(doctorRepository.findById(1L)).thenReturn(Optional.of(doctor));
+        when(appointmentRepository.getBetweenTime(1L, startTime, endTime)).thenReturn(new ArrayList<>());
+        AppointmentDto expectedResult = new AppointmentDto();
```

Comment:
uzywaj moze konstruktora all args wtedy bedzie to przyjemnije wygladalo

Reply:
do zastosowania w wiekszosci testow

Reply:
w przyszłości będzie tu builder